File: Tree/Binary Tree Inorder Traversal.cpp
// TC:O(n) SC:(h) 
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    void fun(TreeNode*& root, vector<int>& ans) {
        // Inorder: Left->root->Right
        if (!root)
            return;
        fun(root->left, ans);
        ans.emplace_back(root->val);
        fun(root->right, ans);
    }
    vector<int> inorderTraversal(TreeNode* root) {
        if (!root)
            return {};
        vector<int> ans;

        fun(root, ans);

        return ans;
    }
};

File: Tree/Python/Binary Tree Inorder Traversal.py
## TC:O(n) SC:O(h)

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def fun(self, root, ans):
        if not root:
            return
        self.fun(root.left, ans)
        ans.append(root.val)
        self.fun(root.right, ans)

    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        self.fun(root, ans)
        return ans
